packages/core-data/src/types.ts

```diff
@@ -0,0 +1,85 @@
+export type Context = 'edit' | 'embed' | 'view';
+
+interface Edit {}
+
+// @ts-ignore
+export interface EntityRecord< C extends Context > {}
+
+export interface EntityConfig< Record extends EntityRecord<any> > {
```

Comment:
I am not sure if parametrizing this will play along with `as const`/`DeepReadOnly` and the lookup table approach. If not, perhaps we could hardcode the key and the context in TypeScript, separately from the config, and then require the `EntityConfig` type to only take these hardcoded values? But that seems backwards, as if the type was the actual config.

Reply:
not sure what I was thinking there with `Record`. nevertheless, we should consider all of this possibly junk-code because I was laser-focused on the other aspects while writing it out.